src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs

```diff
@@ -141,6 +145,60 @@ private void WriteScopeInformation(Utf8JsonWriter writer, IExternalScopeProvider
             }
         }
 
+        private void WriteItem(Utf8JsonWriter writer, KeyValuePair<string, object> item)
+        {
+            var key = item.Key;
+            switch (item.Value)
+            {
+                case bool boolValue:
+                    writer.WriteBoolean(key, boolValue);
+                    break;
+                case byte byteValue:
+                    writer.WriteNumber(key, byteValue);
+                    break;
+                case sbyte sbyteValue:
+                    writer.WriteNumber(key, sbyteValue);
+                    break;
+                case char charValue:
+#if NETCOREAPP
+                    writer.WriteString(key, MemoryMarshal.CreateSpan(ref charValue, 1));
+#else
+                    writer.WriteString(key, charValue.ToString());
+#endif
+                    break;
+                case decimal decimalValue:
+                    writer.WriteNumber(key, decimalValue);
+                    break;
+                case double doubleValue:
+                    writer.WriteNumber(key, doubleValue);
+                    break;
+                case float floatValue:
+                    writer.WriteNumber(key, floatValue);
+                    break;
+                case int intValue:
+                    writer.WriteNumber(key, intValue);
+                    break;
+                case uint uintValue:
+                    writer.WriteNumber(key, uintValue);
+                    break;
+                case long longValue:
+                    writer.WriteNumber(key, longValue);
+                    break;
+                case ulong ulongValue:
+                    writer.WriteNumber(key, ulongValue);
+                    break;
+                case short shortValue:
+                    writer.WriteNumber(key, shortValue);
+                    break;
+                case ushort ushortValue:
+                    writer.WriteNumber(key, ushortValue);
+                    break;
+                default:
+                    writer.WriteString(key, ToInvariantString(item.Value));
```

Comment:
Is there a test that captures when `item.Value` is `null`? Just to prevent a regression in the future.

Reply:
I'm not convinced this is the right place for the test, as a formatter should never get a null value.
We don't have tests at that level so it would require a bit of work to add one but I can do it.
What do you think?
Would a `Debug.Assert(item.Value != null)` be enough instead?

Reply:
The logic is here https://github.com/dotnet/runtime/blob/0f7370baa68c5809f93095cbce9bf614f44cc731/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LogValuesFormatter.cs#L177-L180

It doesn't look like there is a test, however I think it should be left in a separate PR.
Should I log an issue?

Reply:
Perhaps a use case could look like this: 
```
logger.Log(LogLevel.Information, 0, state: new WithNullValue(), exception: null, formatter: (a, b) => string.Empty);
```
Where WithNullValue is:
```

    internal class WithNullValue : IReadOnlyList<KeyValuePair<string, object>>
    {
        int IReadOnlyCollection<KeyValuePair<string, object>>.Count { get; } = 1;

        KeyValuePair<string, object> IReadOnlyList<KeyValuePair<string, object>>.this[int index]
        {
            get
            {
                if (index == 0)
                {
                    return new KeyValuePair<string, object>("KeyForNull", null);
                }

                throw new ArgumentOutOfRangeException(nameof(index));
            }
        }

        IEnumerator<KeyValuePair<string, object>> IEnumerable<KeyValuePair<string, object>>.GetEnumerator()
        {
            yield return new KeyValuePair<string, object>("KeyForNull", null);
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            return ((IEnumerable<KeyValuePair<string, object>>)this).GetEnumerator();
        }
    }
```

Reply:
Ah! Goo one! Thank you!
I always forget that the `Log` method is public.

Reply:
I've added the test and changed the behaviour to serialize as null, instead of empty string, given `null` is a possible value when you don't rely on the extension methods.